rest-api/alembic/versions/71614b7c6482_filter_ghosts_from_views.py

```diff
@@ -0,0 +1,505 @@
+   ps.sample_status_1ed04,
+   ps.sample_status_1ed04_time,
+   ps.sample_status_1ed10,
+   ps.sample_status_1ed10_time,
+   ps.sample_status_2ed10,
+   ps.sample_status_2ed10_time,
+   ps.sample_status_1ur10,
+   ps.sample_status_1ur10_time,
+   ps.sample_status_1sal,
+   ps.sample_status_1sal_time,
+   ps.sample_status_1sal2,
+   ps.sample_status_1sal2_time,
+   ps.sample_status_1cfd9,
+   ps.sample_status_1cfd9_time,
+   ps.sample_status_1pxr2,
+   ps.sample_status_1pxr2_time,
+   ps.num_completed_baseline_ppi_modules,
+   ps.num_completed_ppi_modules,
+   ps.num_baseline_samples_arrived,
+   ps.samples_to_isolate_dna,
+   ps.consent_for_cabor,
+   ps.consent_for_cabor_time,
+   (SELECT IFNULL(GROUP_CONCAT(
+      IF(ac.value = 'WhatRaceEthnicity_RaceEthnicityNoneOfThese',
+         'NoneOfThese',
+         TRIM(LEADING 'WhatRaceEthnicity_' FROM
+              TRIM(LEADING 'PMI_' FROM ac.value)))),
+    'None')
+    FROM questionnaire_response qr, questionnaire_response_answer qra,
+      questionnaire_question qq, code c, code ac
+    WHERE qra.end_time IS NULL AND
+          qr.questionnaire_response_id = qra.questionnaire_response_id AND
+          qra.question_id = qq.questionnaire_question_id AND
+          qq.code_id = c.code_id AND c.value = 'Race_WhatRaceEthnicity' AND
+          qr.participant_id = p.participant_id AND
+          qra.value_code_id = ac.code_id AND
+          ac.value != 'WhatRaceEthnicity_Hispanic'
+   ) race_codes,
+   (SELECT COUNT(ac.value)
+    FROM questionnaire_response qr, questionnaire_response_answer qra,
+      questionnaire_question qq, code c, code ac
+    WHERE qra.end_time IS NULL AND
+      qr.questionnaire_response_id = qra.questionnaire_response_id AND
+      qra.question_id = qq.questionnaire_question_id AND
+      qq.code_id = c.code_id AND c.value = 'Race_WhatRaceEthnicity' AND
+      qr.participant_id = p.participant_id AND
+      qra.value_code_id = ac.code_id AND
+      ac.value = 'WhatRaceEthnicity_Hispanic'
+   ) hispanic
+ FROM
+   participant p
+     LEFT OUTER JOIN hpo ON p.hpo_id = hpo.hpo_id
+     LEFT OUTER JOIN participant_summary ps ON p.participant_id = ps.participant_id
+     LEFT OUTER JOIN code state_code ON ps.state_id = state_code.code_id
+     LEFT OUTER JOIN code recontact_method_code ON ps.recontact_method_id = recontact_method_code.code_id
+     LEFT OUTER JOIN code language_code ON ps.language_id = language_code.code_id
+     LEFT OUTER JOIN code gender_code ON ps.gender_identity_id = gender_code.code_id
+     LEFT OUTER JOIN code sex_code ON ps.sex_id = sex_code.code_id
+     LEFT OUTER JOIN code sexual_orientation_code ON ps.sexual_orientation_id = sexual_orientation_code.code_id
+     LEFT OUTER JOIN code education_code ON ps.education_id = education_code.code_id
+     LEFT OUTER JOIN code income_code ON ps.income_id = income_code.code_id
+     WHERE p.withdrawal_status = 1 AND # NOT_WITHDRAWN
+           (ps.email IS NULL OR ps.email NOT LIKE '%@example.com') AND
```

Comment:
Does comment allowed in """ """ ?

Reply:
yes if it's raw sql it's fine and distinguishes between any `'` and `"` quotes that may be in the query.